frontends/p4/validateParsedProgram.cpp

```diff
@@ -110,15 +114,17 @@ void ValidateParsedProgram::distinctParameters(
     for (auto p : apply->parameters) {
         auto it = found.find(p->getName());
         if (it != found.end())
-            ::error("Duplicated parameter name: %1% and %2%",
+            ::error(ErrorType::ERR_INVALID,
+                    "name.  Duplicates %2%",
                     it->second, p);
         else
             found.emplace(p->getName(), p);
     }
     for (auto p : constr->parameters) {
         auto it = found.find(p->getName());
         if (it != found.end())
-            ::error("Duplicated parameter name: %1% and %2%",
+            ::error(ErrorType::ERR_INVALID,
+                    "name. Duplicates %2%",
```

Comment:
Since this is the same name, it would be good to find a way in which we can distinguish between the current name and its duplicate. In the current error we mark only one of the 'p's:
```
dup-param1.p4(17): [--Werror=invalid] error: p: Invalid name.  Duplicates p
control MyIngress<p>(inout bit<32> p) {
                  ^
```
Same for the other duplicates.